core/src/main/java/jenkins/security/ApiTokenProperty.java

```diff
@@ -358,6 +359,38 @@ public ApiTokenStats getTokenStats() {
         return tokenStats;
     }
 
+    // essentially meant for scripting
+    @Restricted(NoExternalUse.class)
+    public @NonNull String addFixedNewToken(@NonNull String name, @NonNull String tokenPlainValue) throws IOException {
+        String tokenUuid = this.tokenStore.addFixedNewToken(name, tokenPlainValue);
+        user.save();
+        return tokenUuid;
+    }
+    
+    // essentially meant for scripting
+    @Restricted(NoExternalUse.class)
+    public @NonNull TokenUuidAndPlainValue generateNewToken(@NonNull String name) throws IOException {
+        TokenUuidAndPlainValue tokenUuidAndPlainValue = tokenStore.generateNewToken(name);
+        user.save();
+        return tokenUuidAndPlainValue;
+    }
+    
+    // essentially meant for scripting
+    @Restricted(NoExternalUse.class)
+    public void revokeAllTokens() throws IOException {
+        tokenStats.removeAll();
+        tokenStore.revokeAllTokens();
+        user.save();
+    }
+    
+    // essentially meant for scripting
+    @Restricted(NoExternalUse.class)
+    public void revokeAllTokensExceptOne(@NonNull String tokenUuid) throws IOException {
```

Comment:
any reason not to add a revoke individual token as well to complete the API?

Reply:
@timja You already have the endpoint ApiTokenProperty#doRevoke for singular tokenUuid, and so I will add script equivalent :)